### PR TITLE
Upsert by unique key

### PR DESCRIPTION
**Pull request by IvanHdzC**
This PR fixes issue for upsert transaction. Now a days, the transaction used to be one for all processed batch, now it's one for each different unique key. This Unique key is the one specified on the agent conf.

### CODE REVIEW

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/sql/SQLQueryUtilsTest.java [hunk 1]
```diff
@@ -269,12 +267,10 @@ public void testMySQLUpsertQueryMultipleBatch() {
                 sqlInstance,
                 destination,
                 attrNativeTypes);
-
+        sqlupsertQuery = upsertList.get(1);
```
Comment (AlvaroVega) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/sql/SQLQueryUtilsTest.java:270]: Is safe access to element by 1 (i.e. only 1 element is in ursertList)?

Comment (AlvaroVega) [reply cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/sql/SQLQueryUtilsTest.java:270]: NTC, it is just a test

**Review by AlvaroVega — APPROVED** [review]
LGTM

**Review by fgalan — APPROVED** [review]
LGTM

**Pull request merged**